lib/handlers/router-entities/route-caching/dynamo-route-caching-provider.ts

```diff
@@ -167,14 +167,24 @@ export class DynamoRouteCachingProvider extends IRouteCachingProvider {
   ): CachedRoutes {
     metric.putMetric(`RoutesDbEntriesFound`, result.Items!.length, MetricLoggerUnit.Count)
     const cachedRoutesArr: CachedRoutes[] = result.Items!.map((record) => {
-      // If we got a response with more than 1 item, we extract the binary field from the response
-      const itemBinary = record.item
-      // Then we convert it into a Buffer
-      const cachedRoutesBuffer = Buffer.from(itemBinary)
-      // We convert that buffer into string and parse as JSON (it was encoded as JSON when it was inserted into cache)
-      const cachedRoutesJson = JSON.parse(cachedRoutesBuffer.toString())
-      // Finally we unmarshal that JSON into a `CachedRoutes` object
-      return CachedRoutesMarshaller.unmarshal(cachedRoutesJson)
+      if (record.plainRoutes?.toString().trim() !== '') {
```

Comment:
i believe you also want to check if `record.plainRoutes` is defined
`if (record.plainRoutes && record.plainRoutes.toString().trim() !== '')` otherwise if plainRoutes is undefined/null it will enter

Reply:
big +1, especially on the test front

Reply:
added a test to assert it goes through the plaintext routes path, with the metrics logging

Reply:
i would also add a test for `record.plainRoutes = undefined/null/empty string` to make sure binary path is called.

Reply:
ok, figured out a way to update the test setup to invoke the binary path. Basically I in-place update the cached routes plainRoutes column to be undefined, so that it hits the binary path. Since I assert that `RoutesDbEntrySerializedRouteFound` metric gets logged, it must have hit the binary path.